pkg/console/operator/sync_v400.go

```diff
@@ -114,44 +113,14 @@ func sync_v400(co *consoleOperator, consoleConfig *v1alpha1.Console) (*v1alpha1.
 func SyncDeployment(co *consoleOperator, recorder events.Recorder, consoleConfig *v1alpha1.Console, cm *corev1.ConfigMap, serviceCAConfigMap *corev1.ConfigMap, sec *corev1.Secret) (*appsv1.Deployment, bool, error) {
 	logrus.Printf("validating console deployment...")
 	requiredDeployment := deploymentsub.DefaultDeployment(consoleConfig, cm, serviceCAConfigMap, sec)
-	versionAvailability := []operatorv1.GenerationStatus{
-		{
-			Group:          "apps/v1",
-			Resource:       "deployments",
-			Namespace:      requiredDeployment.Namespace,
-			Name:           requiredDeployment.Name,
-			LastGeneration: requiredDeployment.Generation,
-		},
-	}
-	deploymentGeneration := resourcemerge.ExpectedDeploymentGeneration(requiredDeployment, versionAvailability)
-	// first establish, do we have a deployment?
-	existingDeployment, getDepErr := co.deploymentClient.Deployments(api.TargetNamespace).Get(deploymentsub.Stub().Name, metav1.GetOptions{})
-
-	// if not, create it, first pass
-	if apierrors.IsNotFound(getDepErr) {
-		logrus.Print("deployment not found, creating new deployment")
-		_, depCreated, createdErr := resourceapply.ApplyDeployment(co.deploymentClient, recorder, requiredDeployment, deploymentGeneration, true)
-		// kill the sync loop
-		return nil, depCreated, fmt.Errorf("deployment not found, creating new deployment, create error = %v", createdErr)
-	}
-
-	if getDepErr != nil {
-		logrus.Errorf("%q: %v \n", "deployment", getDepErr)
-		return nil, false, getDepErr
-	}
-
-	// otherwise, we may need to update or force a rollout
-	if deploymentsub.ResourceVersionsChanged(existingDeployment, cm, serviceCAConfigMap, sec) {
-		toUpdate := deploymentsub.UpdateResourceVersions(existingDeployment, cm, serviceCAConfigMap, sec)
-		updatedDeployment, depChanged, updateErr := resourceapply.ApplyDeployment(co.deploymentClient, recorder, toUpdate, deploymentGeneration, true)
-		if updateErr != nil {
-			logrus.Errorf("%q: %v \n", "deployment", updateErr)
-			return nil, false, updateErr
-		}
-		return updatedDeployment, depChanged, nil
+	expectedGeneration := getDeploymentGeneration(co)
```

Comment:
So we still have work to do here, with generation, per our conversation. Should we comment to track, or assume a fast follow?

Reply:
Let's open a JIRA issue